qutebrowser/browser/history.py

```diff
@@ -285,17 +148,122 @@ def add_url(self, url, title="", *, redirect=False, atime=None):
             log.misc.warning("Ignoring invalid URL being added to history")
             return
 
-        if atime is None:
-            atime = time.time()
-        entry = Entry(atime, url, title, redirect=redirect)
-        if self._initial_read_done:
-            self._add_entry(entry)
-            self._new_history.append(entry)
-            self.item_added.emit(entry)
-            if not entry.redirect:
-                self.add_completion_item.emit(entry)
+        atime = int(atime) if (atime is not None) else int(time.time())
+        url_str = url.toString(QUrl.FullyEncoded | QUrl.RemovePassword)
+        self.insert({'url': url_str,
+                     'title': title,
+                     'atime': atime,
+                     'redirect': redirect})
+        if not redirect:
+            self.completion.insert({'url': url_str,
+                                    'title': title,
+                                    'last_atime': atime},
+                                   replace=True)
```

Comment:
Hmmm... did you just get into the situation where `flake8` doesn't like this that way, and `pylint` doesn't like it when you add a space? If so, don't worry, I'll make it shut up when merging 😆